src/shared/Input/index.tsx

```diff
@@ -0,0 +1,61 @@
+import { FC, useState, MouseEvent } from 'react';
+import { CustomInputProps } from '@ts/interfaces';
+import { IconButton, InputAdornment } from '@mui/material';
+import OpenedEye from '@assets/input/eye-opened.svg';
+import ClosedEye from '@assets/input/eye-password-hide-svgrepo-com.svg';
+import { StyledInput } from './styled';
+
+const CustomInput: FC<CustomInputProps> = ({
+  text,
+  type,
+  startInputIcon,
+  endInputIcon,
+}) => {
+  const [showPassword, setShowPassword] = useState(false);
+
+  const handleTogglePasswordVisibility = (
+    event: MouseEvent<HTMLButtonElement>
+  ) => {
+    event.preventDefault();
+    setShowPassword((show) => !show);
+  };
+
+  return (
+    <StyledInput
```

Comment:
please rewrite it like this
 
```
 <StyledInput
  placeholder={text}
  type={showPassword ? 'text' : 'password'}
  fullWidth
  InputProps={{
    startAdornment: startInputIcon && <StartInputAdornment />,
    endAdornment:
      type === 'password' ? <PasswordVisibilityToggle /> : (endInputIcon && <EndInputAdornment />),
  }}
/>
```
and move the hidden logic to the new components